queries/futures/useGetAllFuturesTradesForAccount.ts

```diff
@@ -0,0 +1,64 @@
+import { useQuery, UseQueryOptions } from 'react-query';
+import { useRecoilValue } from 'recoil';
+
+import { appReadyState } from 'store/app';
+import { isL2State, isWalletConnectedState, networkState } from 'store/wallet';
+
+import QUERY_KEYS from 'constants/queryKeys';
+import { getFuturesEndpoint, mapTrades } from './utils';
+import { FuturesTrade } from './types';
+import { getFuturesTrades } from './subgraph';
+import { DEFAULT_NUMBER_OF_TRADES } from 'constants/defaults';
+
+const useGetAllFuturesTradesForAccount = (
+	account?: string | null,
+	options?: UseQueryOptions<FuturesTrade[] | null> & { forceAccount: boolean }
+) => {
+	const isAppReady = useRecoilValue(appReadyState);
+	const network = useRecoilValue(networkState);
+	const futuresEndpoint = getFuturesEndpoint(network);
+	const isWalletConnected = useRecoilValue(isWalletConnectedState);
+	const isL2 = useRecoilValue(isL2State);
+
+	return useQuery<FuturesTrade[] | null>(
+		QUERY_KEYS.Futures.AllTradesAccount(network.id, account || null),
+		async () => {
+			if (!account) return null;
+
+			try {
+				const response = await getFuturesTrades(
+					futuresEndpoint,
+					{
+						first: DEFAULT_NUMBER_OF_TRADES,
+						where: {
+							account: account,
+						},
+						orderDirection: 'desc',
+						orderBy: 'timestamp',
+					},
+					{
+						id: true,
+						timestamp: true,
+						account: true,
+						size: true,
+						asset: true,
+						price: true,
+						positionId: true,
+						positionSize: true,
+						positionClosed: true,
+						pnl: true,
+						feesPaid: true,
+						orderType: true,
+					}
+				);
+				return response ? mapTrades(response) : null;
+			} catch (e) {
+				console.log(e);
+				return null;
```

Comment:
Not sure what is the common approach here. All hooks use same approach for error handling. I suppose it requires some discussions regarding how to handle query error.

Reply:
@Tburm @koredefashokun not sure if you have any thoughts on this but I think it would be better to allow the errors to bubble up in query hooks so the correct error state is visible on the query object.

Reply:
This is a good point. So the idea would be that `useQuery` throws the error and we handle it in the component instead?

Reply:
Yeah @avclarke @Tburm. I think we probably should just not catch the error here and let React Query do it. That said, we do need a good way to decode errors present errors in the UI as well (outside the scope of this PR though).

Reply:
@koredefashokun I think we catch a lot of errors in React Query this way, so we should change here but document a greater cleanup effort for that and catching more gracefully when they fail.

Reply:
Yea sounds good, let's make the change here then @joeyzhaozhao and we'll add a new ticket to make this more consistent across the codebase

Reply:
@avclarke, sounds good to me. Just to clarify, since we would no long handle error here try/catch and console.log should be removed, right?

Reply:
yep that's right, can remove try/catch and console.log

Reply:
Done!